lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90VectorReader.java

```diff
@@ -386,9 +387,18 @@ public int nextDoc() {
     }
 
     @Override
-    public int advance(int target) throws IOException {
-      // We could do better by log-binary search in ordToDoc, but this is never used
-      return slowAdvance(target);
+    public int advance(int target) {
+      assert docID() < target;
+      ord = Arrays.binarySearch(fieldEntry.ordToDoc, ord + 1, fieldEntry.ordToDoc.length, target);
+      if (ord < 0) {
+        ord = -(ord + 1);
+      }
+      if (ord == fieldEntry.ordToDoc.length) {
```

Comment:
should this read `ord >= fieldEntry.ordToDoc.length` to make sure we never grow beyond?

Reply:
I think the range of values for `ord` will be between `[0,fieldEntry.ordToDoc.length]`, so we dont need a `>` check ?

Reply:
I guess it could save us from some mistake. We could add an `assert ord <= fieldEntry.ordToDoc.length`

Reply:
sure, will add a check.